Mapped Skin to Profile instead of User and Added editSkin User interface

laurenyew opened the pull request:
- Mapped Skin to Profile instead of User
- Added editSkin User interface (to get to it, go to account, and there is
  a link to the skin page where you can edit your skin's settings.) You
  can either write in the changes and click save, or press enter.

Note: some changes may not be immediately visable (for instance, changing
the parameters for buttons), you need to click the OpenBook link to go
back to the home page to see that change.

Note2: if the HTML does not use the main.css file, it will not get these
parametes (ex: it looks like the font size and type for links are set
elsewhere).

IMPORTANT!!!
Because these changes are not hardcoded into the initial-data.yml, if you
close your play run session, your changes will be lost!

Testing: added unit testing to test/unit/SkinTest.java. Not yet sure
about how to go about testing the user interface.

laurenyew committed:
3/7/12
- Mapped Skin to Profile instead of User
- Added editSkin User interface (to get to it, go to account, and there is
a link to the skin page where you can edit your skin's settings.) You
can either write in the changes and click save, or press enter.

Note: some changes may not be immediately visable (for instance, changing
the parameters for buttons), you need to click the OpenBook link to go
back to the home page to see that change.

Note2: if the HTML does not use the main.css file, it will not get these
parametes (ex: it looks like the font size and type for links are set
elsewhere).

IMPORTANT!!!
Because these changes are not hardcoded into the initial-data.yml, if you
close your play run session, your changes will be lost!

Testing: added unit testing to test/unit/SkinTest.java. Not yet sure
about how to go about testing the user interface.

laurenyew committed:
Adding in skin.html to tracking

amshali commented on app/models/Skin.java:
```diff
@@ -139,6 +138,16 @@ public Skin(String name)
   public void cloneSkin(Skin c)
   {
+   
+   /* 
```
remove the commented code.

jtessler commented on app/controllers/Skins.java:
```diff
@@ -30,30 +30,82 @@ public static void skin(Long id) {
-    user.skin.save();  
+    
+    //body
+    if (given(update.bodyBGColor)) { user.profile.skin.bodyBGColor = update.bodyBGColor; }
```
What happened to our coding conventions?

jtessler commented on test/unit/SkinTest.java:
```diff
@@ -20,31 +20,44 @@ public void testDatabase()
+   */
+  @Test
+  public void testEditSkin()
+  {
```
Why add a blank test?

amshali replied:
?

jtessler commented on app/controllers/Skins.java:
```diff
@@ -30,30 +30,82 @@ public static void skin(Long id) {
+    if (given(update.buttonLinkUnvisitedColor)) { user.profile.skin.buttonLinkUnvisitedColor = update.buttonLinkUnvisitedColor; }
+    if (given(update.buttonLinkVisitedColor)) { user.profile.skin.buttonLinkVisitedColor = update.buttonLinkVisitedColor; }
+    if (given(update.buttonLinkHoverColor)) { user.profile.skin.buttonLinkHoverColor = update.buttonLinkHoverColor; }
+    if (given(update.buttonLinkSelectedColor)) { user.profile.skin.buttonLinkSelectedColor = update.buttonLinkSelectedColor; }
```
There must be a cleaner way to handle this user input. With all of these Skin attributes, what if we just had a `Map<AttributeName,AttributeValue>` instead of all these fields? I picture the number of attributes exploding in size as our page become more complex.

amshali replied:
I like Joe's idea. Please do this.

laurenyew committed:
Setting up map for Skins and their attributes (SkinPair.java).
Problems with editSkin page

laurenyew committed:
Merge branch 'master' of git://github.com/w7cook/OpenBook into laurenyew

Conflicts:
	app/models/Profile.java

laurenyew committed:
Still struggling with how to do inputs on editSkin page.

laurenyew committed:
3/9/12
Finished map of SkinPair attributes to Skin (located in parameters List<SkinPair>
Cleaned up cloning, setting attributes, and creating Skins.
EditSkin page and Skins in main.html should be working (UI tested manually).

laurenyew commented on app/models/Profile.java:
```diff
@@ -12,7 +13,7 @@
   @OneToOne
   public User owner;
-  public String gender; // The user's gender: female or male
+  public String gender; // The user's gender:ï¿½femaleï¿½or male
```
Will pull in fix. For some reason doing pull upstream master gave me this and some weird text.

laurenyew commented on test/unit/BasicTest.java:
```diff
@@ -69,11 +69,11 @@ public void testSigOther() {
 		User alice = new User("alice@gmail.com", "secret", "Alice");
 		alice.save();
-		alice.profile.significant_other = bob;
+		alice.profile.significantOther = bob;
```
significant_other was causing errors

laurenyew committed:
Fixing small problems (strange characters, extra spaces)

laurenyew committed:
Merge branch 'master' of git://github.com/w7cook/OpenBook into laurenyew

laurenyew committed:
Update to current upstream master version

laurenyew committed:
Clean up newline. Update to current version of upstream master
